testing/integration/otel_test.go

```diff
@@ -0,0 +1,295 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/stretchr/testify/require"
+
+	aTesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/testing/tools/estools"
+	"github.com/elastic/elastic-agent/pkg/testing/tools/testcontext"
+	"github.com/elastic/go-elasticsearch/v8"
+)
+
+const fileProcessingFilename = `/tmp/testfileprocessing.json`
+
+var fileProcessingConfig = []byte(`receivers:
+  filelog:
+    include: [ "/var/log/system.log", "/var/log/syslog"  ]
+    start_at: beginning
+
+exporters:
+  file:
+    path: ` + fileProcessingFilename + `
+service:
+  pipelines:
+    logs:
+      receivers: [filelog]
+      exporters:
+        - file`)
+
+var apmProcessingContent = `2023-06-19 05:20:50 ERROR This is a test error message
+2023-06-20 12:50:00 DEBUG This is a test debug message 2
+2023-06-20 12:51:00 DEBUG This is a test debug message 3
+2023-06-20 12:52:00 DEBUG This is a test debug message 4`
+
+var apmOtelConfig = `receivers:
```

Comment:
Nit: these can be const
```suggestion
const apmProcessingContent = `2023-06-19 05:20:50 ERROR This is a test error message
2023-06-20 12:50:00 DEBUG This is a test debug message 2
2023-06-20 12:51:00 DEBUG This is a test debug message 3
2023-06-20 12:52:00 DEBUG This is a test debug message 4`

const apmOtelConfig = `receivers:
```